src/pcapi/core/users/repository.py

```diff
@@ -64,3 +72,53 @@ def get_newly_eligible_users(since: date) -> List[User]:
         )
         .all()
     )
+
+
+def find_favorite_for_offer_and_user(offer_id: int, user_id: int) -> Query:
+    return models.Favorite.query.filter(models.Favorite.offerId == offer_id, models.Favorite.userId == user_id)
+
+
+def get_favorites_for_offers(offer_ids: List[int]) -> List[models.Favorite]:
+    return models.Favorite.query.filter(models.Favorite.offerId.in_(offer_ids)).all()
```

Comment:
On pourrait sûrement fusionner ces 2 fonctions. :)

```suggestion
def get_favorites(offer_ids: List[int], user_id: int = None) -> List[models.Favorite]:
    query = models.Favorite.query.filter(models.Favorite.offerId.in_(offer_ids))
    if user_id:
        query = query.filter_by(userId=user_id)
    return query
```

(Il faudra adapter l'appelant de `get_favorites_for_offers` car la nouvelle fonction renvoie une Query et plus une liste.)

Reply:
Je laisse cette fusion de coté. `find_favorite_for_offer_and_user` devrait disparaitre dans un avenir que l'on espère pas trop lointain et le résultat de la fusion est une fonction plus complexe à lire.